### PR TITLE
PIM-6013: fix simple and multi select cache

### PR DESCRIPTION
**Pull request by juliensnz**
**Description**

With locale specific attributes and if you use a variant group there we a problem with the select and multi select fields.

In this PR we fix the cache management in simple/multiselect fields.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | NA
| Added Behats                      | Y
| Changelog updated                 | N
| Review and 2 GTM                  | N
| Micro Demo to the PO (Story only) | N
| Migration script                  | NA
| Tech Doc                          | NA

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/multi-select-field.js [hunk 1]
```diff
@@ -106,12 +120,20 @@ define(
...
-                            if (null === this.choicePromise) {
+                            var identifiers = AttributeManager.getValue(
+                                this.model.attributes.values,
+                                this.attribute,
+                                UserContext.get('catalogLocale'),
+                                UserContext.get('catalogScope')
+                            ).data;
+
+                            if (null === this.choicePromise || this.promiseIdentifiers !== identifiers) {
                                 this.choicePromise = $.get(choiceUrl, {
```
Comment (pierallard) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/field/multi-select-field.js]: You you will rebase this on 1.5, be careful of this line

**Comment by LaureBro** [comment]
missing changelog

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 2]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+  Scenario: Change a pim_catalog_simpleselect attribute of a variant group
+    Given I set the "English (United States), French (France)" locales to the "mobile" channel
+    And I am on the "simple_select_local_specific" attribute page
+    And I visit the "Values" tab
+    And I create the following attribute options:
+      | Code  |
+      | red   |
+      | blue  |
+      | green |
+    Then I am on the "caterpillar_boots" variant group page
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: When

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 3]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I visit the "Attributes" tab
+    And I visit the "Marketing" group
+    And I add available attributes Simple
+    And I change the "Simple" to "red"
+    And I save the variant group
+    And I should see the flash message "Variant group successfully updated"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: Then

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 4]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I switch the locale to "fr_FR"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: When

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 5]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I change the "Simple" to "blue"
+    And I save the variant group
+    And I should see the flash message "Variant group successfully updated"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: Then

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 6]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    Then I am on the "boot" product page
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: When

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 7]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I visit the "Marketing" group
+    And I switch the scope to "mobile"
+    And I should see the text "[red]"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: Then

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 8]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I switch the locale to "fr_FR"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: When

File: features/enrich/variant-group/edit_variant_group_attribute_value.feature [hunk 9]
```diff
@@ -98,6 +100,33 @@ Feature: Editing attribute values of a variant group also updates products
...
+    And I should see the text "[blue]"
```
Comment (phaseinducer) [inline features/enrich/variant-group/edit_variant_group_attribute_value.feature]: Then

**Pull request merged**